### PR TITLE
Fix category retrive for bestsellers

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Fix category retrive for bestsellers
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/31321
| How to test?  | See https://github.com/PrestaShop/PrestaShop/issues/31321

### CODE REVIEW

**Review by matks — APPROVED** [review]

File: dashproducts.php [hunk 1]
```diff
@@ -221,7 +221,12 @@ public function getTableBestSellers($date_from, $date_to)
             if (!Validate::isLoadedObject($product_obj)) {
                 continue;
             }
-            $category = new Category($product_obj->getDefaultCategory()['id_category_default'], $this->context->language->id);
+
+            $productCategoryId = $product_obj->getDefaultCategory();
+            if (is_array($product_obj->getDefaultCategory()) && isset($product_obj->getDefaultCategory()['id_category_default'])) {
+                $productCategoryId = $product_obj->getDefaultCategory()['id_category_default'];
+            }
+            $category = new Category($productCategoryId, $this->context->language->id);
```
Comment (Hlavtox) [inline dashproducts.php:229]:
```suggestion
            $productCategoryId = $product_obj->getDefaultCategory();
            if (is_array($productCategoryId) && isset($productCategoryId['id_category_default'])) {
                $productCategoryId = $productCategoryId['id_category_default'];
            }
            $category = new Category($productCategoryId, $this->context->language->id);
```
Saves 3 database queries

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Some performance optimization.

**Comment by djoelleuch** [comment]
Hello @mflasquin  , 

There is a return of review , so we can't test this PR . 
Can you please check this  [comment](https://github.com/PrestaShop/dashproducts/pull/50#pullrequestreview-1380913474) 

Thank you !

**Review by florine2623 — APPROVED** [review]
Hello @mflasquin ,

It works as expected !

Before :
Screenshot 2023-04-21 at 12 12 16


After :
Screenshot 2023-04-21 at 12 13 28

It is QA ✅ 
Thanks!

**Comment by Hlavtox** [comment]
Good job @mflasquin and @florine2623! :-)

**Comment by PeNov** [comment]
Hello,
Thank you for this fix.
Error `(PHP Notice:  Trying to access array offset on value of type int in /home/www/domain.cz/www/domain.cz/modules/dashproducts/dashproducts.php on line 224)` from the web server log gone.

But still is present this error `PHP Warning:  Invalid argument supplied for foreach() in /home/www/domain.cz/www/domain.cz/modules/dashproducts/dashproducts.php on line 219`

If I set period from this month 1.5.2023 - 5.5.2023 - works.

If I set period from 2023-04-24 to 2023-05-05 - showing nothing.
image

Thank you
Best regards
PN

**Pull request merged**